playback/src/mixer/alsamixer.rs

```diff
@@ -0,0 +1,76 @@
+use super::AudioFilter;
+use super::{Mixer, MixerConfig};
+use std::error::Error;
+
+use alsa;
+
+#[derive(Clone)]
+pub struct AlsaMixer {
+    config: MixerConfig,
+}
+
+impl AlsaMixer {
+    fn map_volume(&self, set_volume: Option<u16>) -> Result<(u16), Box<Error>> {
+        let mixer = alsa::mixer::Mixer::new(&self.config.card, false)?;
+        let sid = alsa::mixer::SelemId::new(&*self.config.mixer, self.config.index);
+
+        let selem = mixer.find_selem(&sid).expect("Couldn't find SelemId");
```

Comment:
> Couldn't find SelemId
should probably be
> Couldn't find mixer simple element index <self.config.mixer> on <self.config.card>

Reply:
We should also have a more graceful exit for this kind of thing. Nobody needs a backtrace here.

Reply:
In an ideal world we'd just carry on and fall back to the softvol since it's hardly worth crashing the program if you can't find your mixer. Failing that, I guess just do whatever we currently do elsewhere in librespot (which is probably panic... urgh)

Reply:
Whilst I'm not a fan of crashing everywhere, to keep behaviour consistent across librespot, we should probably crash. We crash for far less in other parts of the code anyway.

Reply:
I don't agree with that argument but up to you.

However, you should change the panic message to something like the one I suggested. The current one is too cryptic.

Reply:
Oh, sorry! You have already addressed this one. I was looking at an outdated diff